feat: add PostHog exporter for AI observability

camachoh1 opened the pull request:
Adds a PostHog exporter to send Mastra LLM tracing data to PostHog's AI analytics platform. This addresses community request #7508.

The implementation maps Mastra's span-based observability to PostHog's event model (`$ai_generation`, `$ai_span`), handling token usage, latency tracking, and parent-child relationships. Includes comprehensive test coverage, documentation, and configurable batching for both long-running and serverless environments.

```typescript
// Create PostHog exporter
const posthogExporter = new PosthogExporter({
  apiKey: process.env.POSTHOG_API_KEY!,
  defaultDistinctId: 'default-user',
});

// Create a tool
const weatherTool = createTool({
  id: 'get-weather',
  description: 'Get current weather for a city',
  inputSchema: z.object({
    city: z.string().describe('The city name'),
  }),
  execute: async ({ city }) => ({
    city,
    temperature: 72,
    condition: 'sunny',
  }),
});

// Create agent with tools
const agent = new Agent({
  id: 'weather-agent',
  name: 'Weather Assistant',
  instructions: 'You are a helpful weather assistant.',
  model: 'openai/gpt-4o-mini',
  tools: { weatherTool },
});

// Create Mastra instance with observability
const mastra = new Mastra({
  agents: { 'weather-agent': agent },
  observability: new Observability({
    configs: {
      posthog: {
        serviceName: 'my-app',
        exporters: [posthogExporter],
      },
    },
  }),
});

// Use the agent - traces automatically sent to PostHog

await agent.generate(
      'What is the weather in San Francisco?',
      {
        tracingOptions: {
          metadata: {
            userId: 'test-user-alice',
            sessionId: 'session-001',
          },
        },
      }
    );
```

Co-Authored-By: Richard LoRicco <rtlo9962@gmail.com>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * PostHog AI observability exporter: LLM analytics, token usage, latency/cost tracking, streaming events, privacy mode, and serverless-optimized batching.

* **Documentation**
  * New integration/reference docs and README with configuration, usage examples, privacy/serverless guidance, event mapping, and environment setup.

* **Tests**
  * Comprehensive test suite covering initialization, span lifecycle, mapping, privacy mode, and shutdown.

* **Chores**
  * New package and build/test/lint tooling for the PostHog provider.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

camachoh1 committed:
feat(observability): add initial PostHog exporter implementation

Adds a new observability package for PostHog integration. Includes tracing implementation, tests, and configuration.

camachoh1 committed:
refactor(observability): improve PostHog exporter code organization and add comprehensive test coverage

camachoh1 committed:
feat(observability): finalize PostHog exporter with docs, linting fixes, and changeset

camachoh1 committed:
docs: refactor PostHog exporter documentation for consistency

Streamline PostHog exporter docs and reference to match the concise format used by other exporters (Langfuse, Braintrust, LangSmith, Arize).

camachoh1 committed:
docs: fix PostHog exporter documentation

- Add usage example to changeset
- Fix logLevel default mismatch (INFO vs warn) in reference docs
- Fix misleading enablePrivacyMode comment in README

epinzur reviewed (approved):
Added a few minor comments.  Overall this looks great! Awesome work. Thanks!

camachoh1 committed:
refactor: address PR feedback - remove redundant root check, optimize cleanup, use SDK privacy mode

- Remove redundant parentSpanId check in handleSpanStarted
- Inline span cleanup in handleSpanEnded to avoid extra Map lookup
- Use PostHog SDK's native privacyMode config instead of manual property stripping
- Update tests to verify privacyMode config is passed correctly

camachoh1 committed:
chore: update pnpm lockfile to fix CI

DanielSLew committed:
Merge branch 'main' into feature/posthog-exporter
